app/api/private-cloud/allprojects/route.ts

```diff
@@ -4,17 +4,48 @@ import { PrivateProject } from '@/queries/types';
 import { privateCloudProjects } from '@/queries/private-cloud';
 import formatDate from '@/components/utils/formatdates';
 import { formatFullName } from '@/components/utils/formatFullName';
+import { z } from 'zod';
+import { getServerSession } from 'next-auth/next';
+import { authOptions } from '@/app/api/auth/options';
 
-export async function GET(req: NextRequest): Promise<NextResponse> {
-  const { searchParams } = new URL(req.url);
-  const search = searchParams.get('search') || '';
-  const ministry = searchParams.get('ministry');
-  const cluster = searchParams.get('cluster');
-  const userEmail = searchParams.get('email');
+const searchParamsSchema = z.object({
+  search: z.string().optional().nullable(),
+  ministry: z.string().optional().nullable(),
+  cluster: z.array(z.string()).optional().nullable(),
```

Comment:
is `optional()` necessary?

Reply:
Removing optional results in the following error:
`Error in handler: ZodError: [
  {
    "code": "invalid_type",
    "expected": "array",
    "received": "undefined",
    "path": [
      "cluster"
    ],
    "message": "Required"
  }
]
    at get error [as error] (webpack-internal:///(rsc)/./node_modules/zod/lib/index.mjs:616:31)
    at ZodObject.parse (webpack-internal:///(rsc)/./node_modules/zod/lib/index.mjs:723:22)
    at GET (webpack-internal:///(rsc)/./app/api/private-cloud/allprojects/route.ts:37:55)
    at async /Users/AMNIJJAR/projects/platform-services-registry/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:62609 {
  issues: [
    {
      code: 'invalid_type',
      expected: 'array',
      received: 'undefined',
      path: [Array],
      message: 'Required'
    }
  ],
  addIssue: [Function (anonymous)],
  addIssues: [Function (anonymous)],
  errors: [
    {
      code: 'invalid_type',
      expected: 'array',
      received: 'undefined',
      path: [Array],
      message: 'Required'
    }
  ]
}`